machine/v1beta1/types_alibabaprovider.go

```diff
@@ -0,0 +1,320 @@
+	// +optional
+	InstanceState *string `json:"instanceState,omitempty"`
+
+	// Conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status
+	// +optional
+	Conditions []AlibabaCloudMachineProviderCondition `json:"conditions,omitempty"`
+}
+
+// AlibabaCloudMachineProviderCondition is a condition in a AlibabaCloudMachineProviderStatus
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type AlibabaCloudMachineProviderCondition struct {
+	// Type is the type of the condition.
+	Type ConditionType `json:"type"`
+
+	// Status is the status of the condition.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// LastProbeTime is the last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime"`
+
+	// LastTransitionTime is the last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// Reason is a unique, one-word, CamelCase reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason"`
+
+	// Message is a human-readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message"`
+}
+
+// InstanceTenancy Specifies whether to create the instance on a dedicated host
+// type InstanceTenancy string
+
+// SystemDiskProperties contains the information regarding the system disk including performance, size, name, and category
+type SystemDiskProperties struct {
+	// Category is the category of the system disk.
+	// Valid values:
+	// cloud_essd: ESSD. When the parameter is set to this value, you can use the SystemDisk.PerformanceLevel parameter to specify the performance level of the disk.
+	// cloud_efficiency: ultra disk.
+	// cloud_ssd: standard SSD.
+	// cloud: basic disk.
+	//For non-I/O optimized instances of retired instance types, the default value is cloud. For other instances, the default value is cloud_efficiency.
+	// +optional
+	Category string `json:"category,omitempty"`
+
+	// PerformanceLevel is the performance level of the ESSD used as the system disk. Default value: PL1.
+	// Valid values:
+	//PL0: A single ESSD can deliver up to 10,000 random read/write IOPS.
+	//PL1: A single ESSD can deliver up to 50,000 random read/write IOPS.
+	//PL2: A single ESSD can deliver up to 100,000 random read/write IOPS.
+	//PL3: A single ESSD can deliver up to 1,000,000 random read/write IOPS.
+	//For more information about ESSD performance levels, see ESSDs.
+	// +optional
+	PerformanceLevel string `json:"performanceLevel,omitempty"`
+
+	// Name is the name of the system disk. The name must be 2 to 128 characters in length. It must start with a letter and cannot start with http:// or https://. It can contain letters, digits, colons (:), underscores (_), and hyphens (-).
+	//This parameter is empty by default.
```

Comment:
what does empty mean?

Reply:
This is the literal definition. Disk name will be an empty string.
```
				"Device": "/dev/xvda",
				"DiskChargeType": "PostPaid",
				"DiskId": "d-0xi7kt9u8vf9n0d1951x",
				"DiskName": "",
```

I will update with your previous verbiage.